include/seqan3/alphabet/quality/quality_base.hpp

```diff
@@ -141,25 +141,11 @@ class quality_base : public alphabet_base<derived_type, size, char>
     };
 
     //!\brief Char to rank conversion table.
-    static std::array<rank_type, 256> constexpr char_to_rank
+    static constexpr rank_type char_to_rank(char_type const chr)
     {
-        [] () constexpr
-        {
-            std::array<rank_type, 256> ret{};
-
-            for (int64_t i = std::numeric_limits<char_type>::lowest(); i <= std::numeric_limits<char_type>::max(); ++i)
-            {
-                if (i < derived_type::offset_char)                     // map too-small to smallest possible
-                    ret[static_cast<rank_type>(i)] = 0;
-                else if (i >= derived_type::offset_char + alphabet_size)  // map too-large to highest possible
-                    ret[static_cast<rank_type>(i)] = alphabet_size - 1;
-                else                                                   // map valid range to identity
-                    ret[static_cast<rank_type>(i)] = i - derived_type::offset_char;
-            }
-
-            return ret;
-        }()
-    };
+        int difference = chr - derived_type::offset_char;
```

Comment:
```suggestion
        int64_t difference = chr - derived_type::offset_char;
```

If I assume that `char_type` is `char32_t`, I would subtract two `uint32_t` (actually `uint_least32_t`).


Or would `auto` work?
And clamp needs all three parameters to be of the same type, but if all are signed, it should be fine?
But I think `int` is not generally correct.

Reply:
You are right. I hate these integers implicit conversion things. This one will still throw warnings if we had an architecture where `chr` is unsigned.

I guess

```suggestion
        int64_t difference = static_cast<int64_t>(chr) - static_cast<int64_t>(derived_type::offset_char);
```

would be the most correct version.